### PR TITLE
ci: update go and vim versions

### PR DESCRIPTION
**Pull request by leitzler**
Update go versions to ["go1.17.8", "go1.18"].
Update latest vim version to v8.2.4698.

### CODE REVIEW

File: .github/workflows/docker-based_tests.yml [hunk 1]
```diff
@@ -35,9 +35,9 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-18.04]
-        go_version: ["go1.16.13", "go1.17.6", "go1.18beta1"]
+        go_version: ["go1.17.8", "go1.18"]
```
Comment (mvdan) [inline .github/workflows/docker-based_tests.yml:38]: why not use v1.17.x and v1.18.x, by the way? I see no reason to control the bugfix updates with precision, as they should never break any API or CLI command.

**Review by mvdan — APPROVED** [review]
FYI the same could be said of the OS version, e.g. ubuntu-latest, though that *does* imply big jumps in versions that could potentially break something like an `apt install` command.

File: .github/workflows/docker-based_tests.yml (hunk 1 above)
Comment (leitzler) [reply .github/workflows/docker-based_tests.yml:38]: We push version specific docker images to docker hub (as a manual step) and I'm not sure how well that would play with an "x-version" (see https://hub.docker.com/r/govim/govim/tags).

Regarding OS version, Paul had some comments in govim/govim#1130 where he had tested newer ubuntu and some tests failed.

I'll leave it as is and wrote govim/govim#1153 covering wildcard versions.

Comment (mvdan) [reply .github/workflows/docker-based_tests.yml:38]: SGTM! replied on the new issue.

**Pull request merged**